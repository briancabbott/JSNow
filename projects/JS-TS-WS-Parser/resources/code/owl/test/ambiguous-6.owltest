---program = decl*
decl =
  'function' identifier '(' ((identifier ',')+ identifier)? ')' '{' stmt* '}' : function
stmt = expr
expr =
  number : literal
 .operators prefix
  '-' : negate
 .operators infix flat
  '+' : plus
  '-' : minus
