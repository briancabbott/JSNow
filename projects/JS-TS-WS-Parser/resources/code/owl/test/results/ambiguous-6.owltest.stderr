error: this grammar is ambiguous

. function a ( ) { 1 - 2 } 

  can be parsed in two different ways: as

. function a ( ) { 1            - 2            } 
  |                |            | expr:literal   
  |                expr:literal expr:negate---   
  |                stmt-------- stmt----------   
  decl:function--------------------------------- 
  program--------------------------------------- 

  or as

. function a ( ) { 1            - 2            } 
  |                expr:literal   expr:literal   
  |                expr:minus-----------------   
  |                stmt-----------------------   
  decl:function--------------------------------- 
  program--------------------------------------- 

